src/core/locking.rs

```diff
@@ -37,9 +37,9 @@ impl InputScope for LockingOutput {
             .inner
             .lock()
             .expect("LockingOutput")
-            .new_metric(name, kind);
+            .new_metric(name.clone(), kind);
         let mutex = self.inner.clone();
-        InputMetric::new(move |value, labels| {
+        InputMetric::new(name.join("/"), move |value, labels| {
```

Comment:
Is a conflict possible here? It would be a corner case and I'm not sure whether it's worth of solving.

```rust
bucket.add_name("a").add_name("b").gauge("c");
bucket.gauge("a/b/c");
```